ext/sockets/sockets.c

```diff
@@ -2516,36 +2516,36 @@ PHP_FUNCTION(socket_export_stream)
    Gets array with contents of getaddrinfo about the given hostname. */
 PHP_FUNCTION(socket_addrinfo_lookup)
 {
+	char *service = NULL;
+	size_t service_len;
 	zend_string *hostname, *key;
-	zval *hint, *service, *zhints = NULL;
+	zval *hint, *zhints = NULL;
 
 	struct addrinfo hints, *result, *rp, *res;
 
 	memset(&hints, 0, sizeof(hints));
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "S|za", &hostname, &service, &zhints) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "S|sa", &hostname, &service, &service_len, &zhints) == FAILURE) {
 		RETURN_NULL();
 	}
 
 	if (zhints) {
 		ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(zhints), key, hint) {
 			if (key) {
-				if (strcmp(ZSTR_VAL(key), "ai_flags") == 0) {
-					hints.ai_flags = Z_LVAL_P(hint);
-				} else if (strcmp(ZSTR_VAL(key), "ai_socktype") == 0) {
-					hints.ai_socktype = Z_LVAL_P(hint);
-				} else if (strcmp(ZSTR_VAL(key), "ai_protocol") == 0) {
-					hints.ai_protocol = Z_LVAL_P(hint);
-				} else if (strcmp(ZSTR_VAL(key), "ai_family") == 0) {
-					hints.ai_family = Z_LVAL_P(hint);
+				if (zend_string_equals_literal(key, "ai_flags")) {
+					hints.ai_flags = zval_get_long(hint);
+				} else if (zend_string_equals_literal(key, "ai_socktype")) {
+					hints.ai_socktype = zval_get_long(hint);
+				} else if (zend_string_equals_literal(key, "ai_protocol")) {
+					hints.ai_protocol = zval_get_long(hint);
+				} else if (zend_string_equals_literal(key, "ai_family")) {
+					hints.ai_family = zval_get_long(hint);
```

Comment:
Wondering if it would be good to throw a warning if ta passed hint is not recognized. Is the list of supported keys here something that is likely to change, or pretty much static?